### PR TITLE
deps,lint: update eslint config

### PR DESCRIPTION
**Pull request by nechaido**
Also fix all linter errors produced by `implicit-arrow-linebreak` rule.

### CODE REVIEW

**Comment by lundibundi** [comment]
What was the reason behind `implicit-arrow-linebreak` rule? I see no harm in having 2 styles of arrow functions, furthermore, I personally prefer to have it without parens or just with curly brackets. Can we remove it from our config?

**Comment by nechaido** [comment]
@lundibundi we can override the rule in our config, but I would prefer to have the same code style throughout the whole organisation, thus the whole idea of `eslint-config-metarhia`.

**Comment by lundibundi** [comment]
@nechaido well, I thought to override it in `eslint-config-metarhia`, that's why I /cc'd @aqrln as he was the one who added it there. I just wanted to discuss it here before creating an issue/changing it there as I have missed that change when it was committed.

**Comment by belochub** [comment]
@lundibundi, there is a way to not use parentheses with this rule, we just have to make sure that function body begins on the same line as it's description, so that '`=>`' is not immediately followed by a line break.

**Comment by aqrln** [comment]
I added the rule when it became available in ESLint because that had already been the house style throughout the org, with JSTP, probably, being the only exception, and I'm pretty sure @tshemsedinov was in favor of this style strongly.

**Comment by aqrln** [comment]
That said, I personally have no strong opinion one way or the other, and consistency within the org was the sole motivation. Today I'd rather just use Prettier, tbh.

File: test/node/connection-inspect.js [hunk 1]
```diff
@@ -26,8 +26,11 @@ test.afterEach((done) => {
 
 test.test('must process inspect messages', (test) => {
   const expectedInterfaces = Object.keys(app.interfaces);
-  const expectedTests = expectedInterfaces.reduce((tests, iface) =>
-    tests + Object.keys(app.interfaces[iface]).length + 1, 1);
+  const expectedTests = expectedInterfaces.reduce(
+    (tests, iface) => (
```
Comment (aqrln) [inline test/node/connection-inspect.js]: and to this one

File: test/node/connection-inspect.js [hunk 2]
```diff
@@ -26,8 +26,11 @@ test.afterEach((done) => {
...
+      tests + Object.keys(app.interfaces[iface]).length + 1
+    ), 1
```
Comment (aqrln) [inline test/node/connection-inspect.js]: Nit: could you please also move the second argument to a new line?

File: test/node/connection-inspect.js [hunk 3]
```diff
@@ -38,9 +41,9 @@ test.test('must process inspect messages', (test) => {
     expectedInterfaces.forEach((iface) => {
       connection.inspectInterface(iface, (error, methods) => {
         test.assertNot(error, `must inspect ${iface}`);
-        Object.keys(app.interfaces[iface]).forEach(method =>
+        Object.keys(app.interfaces[iface]).forEach(method => (
```
Comment (aqrln) [inline test/node/connection-inspect.js]: Ditto, I agree with the linter on this one. Either this, or the function itself should be moved to a new line if we allow line-breaking arrow functions without parens. I find the left-hand side of the diff a bit weird to read.

File: test/node/connection-inspect.js [hunk 4]
```diff
@@ -57,10 +60,10 @@ test.test('must generate remote api after inspect', (test) => {
 
       expectedInterfaces.forEach((iface) => {
         test.assert(iface in api, `api must include '${iface}'`);
-        Object.keys(app.interfaces[iface]).forEach(method =>
+        Object.keys(app.interfaces[iface]).forEach(method => (
```
Comment (aqrln) [inline test/node/connection-inspect.js]: Ditto there and everywhere else this pattern might occur.

File: lib/server.js [hunk 5]
```diff
@@ -234,9 +234,9 @@ class Server {
     }
 
     if (sessionStorageProvider.isAsync()) {
-      sessionStorageProvider.get(sessionId, session =>
+      sessionStorageProvider.get(sessionId, session => (
```
[the remark this answers is not in the document]
Comment (lundibundi) [reply lib/server.js]: imo it's even better to have it in just 2 lines (no additional pattern and put the closing pattern on the same line as body)

[the remark this answers is not in the document]
Comment (aqrln) [reply lib/server.js]: that works for me too, yes

**Comment by nechaido** [comment]
@lundibundi @aqrln PTAL.

File: lib/socket.js [hunk 6]
```diff
@@ -109,28 +109,29 @@ const socketFactory = (connect) => {
...
+    );
 
 // Same as newConnectFn but will instead perform reconnect on existing
 // connection.
 //
 // see transportCommon.newReconnectFn
 //
-const newReconnectFn = connect =>
-  transportCommon.newReconnectFn(socketFactory(connect), Transport);
+const newReconnectFn =
```
Comment (belochub) [inline lib/socket.js]: You have enough space left to start function definition on the same line as you did on line 112 in this file.

File: lib/ws-browser.js [hunk 7]
```diff
@@ -123,10 +123,13 @@ const reconnect =
...
-    connectAndInspect(app, client, interfaces, url, callback),
-  reconnect: (connection, url, callback) =>
-    reconnect(connection, url, callback),
+  connect: (app, client, url, callback) => connect(app, client, url, callback),
+  connectAndInspect:
+    (app, client, interfaces, url, callback) => connectAndInspect(
+      app, client, interfaces, url, callback
+    ),
+  reconnect:
+    (connection, url, callback) => reconnect(
```
Comment (belochub) [inline lib/ws-browser.js]: You can fit complete call to `reconnect` on this line.

File: lib/tls.js [hunk 8]
```diff
@@ -65,12 +65,11 @@ const reconnect = sock.newReconnectFn(tls.connect);
...
-  connect: (app, client, ...options) =>
-    connect(app, client, ...options),
-  connectAndInspect: (app, client, interfaces, ...options) =>
-    connectAndInspect(app, client, interfaces, ...options),
-  reconnect: (connection, ...options) =>
-    reconnect(connection, ...options),
+  createServer: (options, listener) => createServer(options, listener),
+  connect: (app, client, ...options) => connect(app, client, ...options),
+  connectAndInspect:
+  (app, client, interfaces, ...options) => connectAndInspect(
```
Comment (belochub) [inline lib/tls.js]: Shouldn't indentation increase here, like everywhere else?

File: lib/server.js [hunk 9]
```diff
@@ -234,8 +234,8 @@ class Server {
     }
 
     if (sessionStorageProvider.isAsync()) {
-      sessionStorageProvider.get(sessionId, session =>
-        getSessionCallback(session, null, callback)
+      sessionStorageProvider.get(sessionId,
```
Comment (aqrln) [inline lib/server.js]: Can you please move `sessionId` to a new line?

File: lib/ws-internal.js [hunk 10]
```diff
@@ -149,8 +149,10 @@ module.exports = {
   // see transportCommon.newConnectFn
   //   webSocketConfig - web socket client configuration
   //                     (see connectionFactory)
-  connect: (app, client, webSocketConfig, ...options) =>
-    connect(app, client, webSocketConfig, ...options),
+  connect:
+    (app, client, webSocketConfig, ...options) => connect(
+      app, client, webSocketConfig, ...options
+    ),
```
Comment (lundibundi) [inline lib/ws-internal.js]: Maybe just use {} (here and a few other places)? This looks much better imo.
```
connect: (app, client, webSocketConfig, ...options) => { 
  connect(app, client, webSocketConfig, ...options);
}

**Comment by belochub** [comment]
Landed in https://github.com/metarhia/jstp/commit/f530d5c076371b5acbe449103f68022de2a37e23.

**Pull request closed without merging**